server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotShutdownIT.java

```diff
@@ -0,0 +1,530 @@
+                l -> client().admin().cluster().getSnapshots(new GetSnapshotsRequest(repoName), l)
+            )
+        ).getSnapshots();
+        assertThat(snapshots, hasSize(1));
+        assertEquals(SnapshotState.SUCCESS, snapshots.get(0).state());
+
+        if (randomBoolean()) {
+            internalCluster().stopNode(originalNode);
+        }
+
+        safeAwait(SubscribableListener.<ActionResponse.Empty>newForked(l -> {
+            final var clearVotingConfigExclusionsRequest = new ClearVotingConfigExclusionsRequest();
+            clearVotingConfigExclusionsRequest.setWaitForRemoval(false);
+            client().execute(ClearVotingConfigExclusionsAction.INSTANCE, clearVotingConfigExclusionsRequest, l);
+        }));
+
+        clearShutdownMetadata(internalCluster().getCurrentMasterNodeInstance(ClusterService.class));
+    }
+
+    public void testRemoveNodeDuringSnapshotWithOtherRunningShardSnapshots() throws Exception {
+        // SnapshotInProgressAllocationDecider only considers snapshots having shards in INIT state, so a single-shard snapshot such as the
+        // one in testRemoveNodeDuringSnapshot will be ignored when the shard is paused, permitting the shard movement. This test verifies
+        // that the shard is permitted to move even when the snapshot has other shards in INIT state.
+
+        final var repoName = randomIdentifier();
+        createRepository(repoName, "mock");
+
+        // create another index on another node which will be blocked (remain in state INIT) throughout
+        final var otherNode = internalCluster().startDataOnlyNode();
+        final var otherIndex = randomIdentifier();
+        createIndexWithContent(otherIndex, indexSettings(1, 0).put(REQUIRE_NODE_NAME_SETTING, otherNode).build());
+        blockDataNode(repoName, otherNode);
+
+        final var nodeForRemoval = internalCluster().startDataOnlyNode();
+        final var indexName = randomIdentifier();
+        createIndexWithContent(indexName, indexSettings(1, 0).put(REQUIRE_NODE_NAME_SETTING, nodeForRemoval).build());
+
+        final var replacementNode = internalCluster().startDataOnlyNode();
+
+        final var clusterService = internalCluster().getCurrentMasterNodeInstance(ClusterService.class);
+        final var snapshotFuture = startFullSnapshotBlockedOnDataNode(randomIdentifier(), repoName, nodeForRemoval);
+        final var snapshotPausedListener = createSnapshotPausedListener(clusterService, repoName, indexName);
+        waitForBlock(otherNode, repoName);
+
+        putShutdownForRemovalMetadata(nodeForRemoval, clusterService);
+        unblockNode(repoName, nodeForRemoval); // lets the shard snapshot abort, which frees up the shard to move
+        safeAwait(snapshotPausedListener);
+
+        // adjust the allocation filter so that the shard moves
+        updateIndexSettings(Settings.builder().put(REQUIRE_NODE_NAME_SETTING, replacementNode), indexName);
```

Comment:
Is there any specific reason to not use `putNull` here? I think it's better to let shutdown naturally moves the shard instead of forcing it with index settings because the former is more like a realistic processs?

Reply:
Yes I think that makes sense, see b58ba611157.